CMakeLists.txt

```diff
@@ -1,54 +1,102 @@
 cmake_minimum_required(VERSION 3.5)
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_C_STANDARD 99)
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
 project(Fossilize LANGUAGES CXX C)
 
-option(FOSSILIZE_MSVC_ANALYZE "Enable /analyze" OFF)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+set(CMAKE_THREA_PREFER_PTHREAD ON)
+
+include(CMakeDependentOption)
+include(GNUInstallDirs)
+include(CTest)
+
+find_package(Threads REQUIRED)
+find_package(Vulkan)
+
+if (NOT DEFINED CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 14)
```

Comment:
Doesn't this have to be defined before project() to take affect?

Reply:
Surprisingly no! You can test this yourself manually via `get_property(standard TARGET <target-name> PROPERTY CXX_STANDARD)`. CMake uses `CMAKE_CXX_STANDARD` to initialize the `CXX_STANDARD` property on targets, but if it's left blank, it leaves it blank as well and lets the compiler's default kick in. This unfortunately causes problems if you just want to use `/std:latest` on MSVC (as some CMake versions don't understand it), but in practice it's rarely needed :)

Reply:
Ok. As long as C++14 is enabled, I don't really mind how this is solved.

Reply:
Totally understand. 🙂